src/twisted/internet/defer.py

```diff
@@ -1636,40 +1747,56 @@ def loadData(url):
     """
 
     @wraps(f)
-    def unwindGenerator(*args, **kwargs):
+    def unwindGenerator(*args: object, **kwargs: object) -> Deferred:
         try:
             gen = f(*args, **kwargs)
         except _DefGen_Return:
             raise TypeError(
                 "inlineCallbacks requires %r to produce a generator; instead"
                 "caught returnValue being used in a non-generator" % (f,)
             )
-        if not isinstance(gen, types.GeneratorType):
+        # type note: FIXME: error suggests type of gen is wrong:
+        #     Subclass of "Generator[Deferred, object, None]" and "GeneratorType"
+        #     cannot exist: would have incompatible method signatures
+        if not isinstance(gen, GeneratorType):  # type: ignore[unreachable]
             raise TypeError(
                 "inlineCallbacks requires %r to produce a generator; "
                 "instead got %r" % (f, gen)
             )
-        return _cancellableInlineCallbacks(gen)
+        return _cancellableInlineCallbacks(gen)  # type: ignore[unreachable]
 
     return unwindGenerator
 
 
 ## DeferredLock/DeferredQueue
 
 
-class _ConcurrencyPrimitive:
-    def __init__(self):
-        self.waiting = []
+class _ConcurrencyPrimitive(ABC):
+    def __init__(self) -> None:
+        self.waiting = []  # type: List[Deferred]
 
-    def _releaseAndReturn(self, r):
+    def _releaseAndReturn(self, r: _T) -> _T:
         self.release()
         return r
 
-    def run(*args, **kwargs):
+    # You might wonder: "WTF is dbjAAcAiBbiFeBiEAcjgDhfFchjEBahF?"
+    # It is:
+    #     "".join([
+    #         string.ascii_lowercase[int(c)] if c in string.digits else c
+    #         for c in "319AA2A8B18F4B8EA296D75F279EB07F"
+    #     ])
+    # …which is to say that it is a UUID encoded as letters.
+    # …which is to say: "it's not a string that will ever be used as a name in kwargs"
+    def run(
+        dbjAAcAiBbiFeBiEAcjgDhfFchjEBahF,
+        f: Callable[..., object],
+        *args: object,
+        **kwargs: object
+    ) -> Deferred:
```

Comment:
OK perhaps this looks like a weird change.

The tests [`OtherPrimitivesTests.testLock`](https://github.com/twisted/twisted/blob/10017-mypy-defer/src/twisted/test/test_defer.py#L2131) and [`OtherPrimitivesTests.testSemaphore`](https://github.com/twisted/twisted/blob/10017-mypy-defer/src/twisted/test/test_defer.py#L2191) pass in `self=` to this method, which presumably means that `kwargs` here needs to accept any argument, hence the original method signature, which lacked `self` entirely, and the funky argument handling removed below to get it back.

That, of course, makes it hard for a tool like `mypy` to know what's going on with the method signature.  So here we put both `self` and the required first argument (which we can now type as a `Callable`) back in, but then we have to deal with the need to be able to pass `self` in as a keyword to the callable…

So we change the name of `self`, and we want that name not to be some other thing you might want to pass into the callable.  So I took a UUID and encoded it to letters.  If someone ends up trying to pass `dbjAAcAiBbiFeBiEAcjgDhfFchjEBahF` as a keyword here then I HATE THAT JERK SHUT UP GO AWAY.  So there.

Reply:
You could use a positional only argument, or `_v_`

Reply:
Posargs are Py3.8+. I'll add a comment that it would be a better alternative so that when we eventually deprecate Py37-, we can use that instead.

By `_v_`, I assume you mean that as a literal name.  I also thought about `__self` or similar, but went with the above to make it basically impossible for there to be a conflict.  Perhaps overkill, but it's done and doesn't cost anything other than some cognitive load fo someone reading the code, which the comment hopefully abates.

Reply:
Might I suggest the slightly-less-elaborate name `self_319AA2A8B18F4B8EA296D75F279EB07F` ?

Reply:
Also, this sounds like it ought to be a bug on mypy?

Reply:
> Also, this sounds like it ought to be a bug on mypy?

Why? The error from mypy was in the test that passed in `self=`, and the warning was that `self` is being passed in twice, which is correct, I think.

Reply:
> Might I suggest the slightly-less-elaborate name `self_319AA2A8B18F4B8EA296D75F279EB07F` ?

Sure, I don't care what the color of the bike shed ends up being.

Reply:
> Why? The error from mypy was in the test that passed in `self=`, and the warning was that `self` is being passed in twice, which is correct, I think.

Ah, I misunderstood, I thought that that actually worked at runtime given that the test was doing it. I see from a quick experiment that it doesn't, so, nevermind!